### ISSUE TITLE
Documentation is incorrect for setting custom user attribute on 14.0.0. 

### ISSUE THREAD

**Issue by victorraft** [Appboy/appboy-android-sdk#194]
Documentation says that we should do this for setting a custom user attribute: `Appboy.getInstance(context).currentUser?.setCustomUserAttribute("your_attribute_key", "your_attribute_value")`
In v14.0.0 this does not compile, though, because `currentUser` does not exist. 
Could you fix the documentation, please?

**Comment by radixdev** [comment]
Hi @victorraft ,

This is actually a Kotlin issue with how we defined the `getCurrentUser()` method and introduced generics for it. We will fix this in a new release and I'll update this issue once it's live.

**Comment by victorraft** [comment]
Thanks for the response @radixdev !
Do you have an ETA on when this fix should be released? 
Is there a recommended workaround that I could use for now?

**Comment by radixdev** [comment]
It would be a breaking change (`AppboyUser` wouldn't directly be allowed) so sometime within the month most likely.

Also as a preface, `Appboy.getInstance(context).currentUser` without using the callback is a blocking call so I would recommend the following as a long term, non blocking, replacement:

First an extension 
```
fun Braze.runOnUser(block: (user:BrazeUser) -> Unit) {
  this.getCurrentUser(object : SimpleValueCallback<BrazeUser>() {
    override fun onSuccess(user: BrazeUser) {
      super.onSuccess(user)
      block(user)
    }
  })
}
```

Then it's called via:

```
Braze.getInstance(context).runOnUser { user -> user.setCustomUserAttribute("your_attribute_key", "your_attribute_value") }
``` 

We're thinking of releasing a standalone kotlin extensions module that would contain extension functions like the above. If you have any other Kotlin related suggestions or issues (no matter how minor), please bring them up!

**Comment by victorraft** [comment]
I guess it would be nice to have a suspending way of getting the user. So maybe something like transforming the callback based api into a coroutine based api, like that:

```
    suspend fun Braze.getCurrentUserSuspending(): BrazeUser? = suspendCoroutine { continuation ->
        this.getCurrentUser(object : IValueCallback<BrazeUser> {
            override fun onSuccess(user: BrazeUser) {
                continuation.resume(user)
            }

            override fun onError() {
                continuation.resume(null)
            }
        })
    }
```

then it can be called like this:

```
    suspend fun setAttribute() {
        Braze.getInstance(context).getCurrentUserSuspending()?.setCustomUserAttribute("your_attribute_key", "your_attribute_value") 
    }
```

**Comment by victorraft** [comment]
@radixdev would you say that one of the approaches above (either runOnUser or getCurrentUserSuspending) would work for 14.0.0? Can I use them? Just asking to make sure, since this is not in the documentation :)

**Comment by radixdev** [comment]
Yes either would work since the type is present

**Comment by victorraft** [comment]
Ok, thank you very much! Please let me know when the next version is released!

**Comment by victorraft** [comment]
Ah, one more question: should I call getCurrentUser everytime I need to set a custom attribute? Or can I call it only once and then save the result and use it in all the next setCustomUserAttribute calls?

**Comment by radixdev** [comment]
I would call it every time to account for if you call change user or disable SDK

**Comment by radixdev** [comment]
Hi @victorraft ,

`Appboy.getInstance(this).getCurrentUser<BrazeUser>()` would also work. We'll be updating our documentation to not have the outdated examples.

**Comment by milosmns** [comment]
+1 for updating the docs to reflect undocumented breaking changes.

**Comment by milosmns** [comment]
One other docs-related question - I see that ```com.appboy.AppboyLifecycleCallbackListener``` is still being used to register Activity lifecycle callbacks, whereas most of the other components have been renamed properly to `Braze*`. 

Is it ok to keep using Appboy's listener?

**Comment by radixdev** [comment]
Hi @milosmns ,

We just put up a PR to update our docs on this 👍🏾 .

It is also ok to keep using that listener. We will most likely mark the Appboy class as deprecated and suggest the Braze version in a coming SDK update. Both will be available for a long time.

**Comment by radixdev** [comment]
Our documentation update is now live, thanks for filing.
